src/components/Avatar.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <div v-if="enabled">
+    <oc-spinner v-if="loading" />
+    <oc-avatar width=42 height=42 v-if="!loading && avatarSource !== ''" :src="avatarSource" />
+    <oc-icon size="large" v-if="!loading && avatarSource === ''" name="account_circle" color="white" class="uk-display-inline" />
+  </div>
+</template>
+<script>
+import { mapGetters } from 'vuex'
+
+export default {
+  data () {
+    return {
+      /**
+       * Set to object URL when loaded, or on failure, icon placeholder is shown
+       */
+      avatarSource: '',
+      loading: true
+    }
+  },
+  mounted: function () {
+    if (!this.enabled) {
+      return;
+    }
+    let headers = new Headers()
+    let instance = this.$root.config.server || window.location.origin
+    let url = instance + '/remote.php/dav/avatars/' + this.userid + '/128.png'
```

Comment:
at time of mounting userid can not yet be set - better move this into a watcher on the property and load upon change